libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs

```
@@ -61,31 +59,52 @@ public static CreateMessageOptions ActivityToTwilio(Activity activity, string tw
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static async Task WriteAsync(HttpResponse response, int code, string text, Encoding encoding, CancellationToken cancellationToken)
         {
-            if (httpRequest == null)
+            if (response == null)
             {
-                return null;
+                throw new ArgumentNullException(nameof(response));
             }
 
-            Dictionary<string, string> body;
-            using (var bodyStream = new StreamReader(httpRequest.Body))
+            if (text == null)
             {
-                body = QueryStringToDictionary(await bodyStream.ReadToEndAsync().ConfigureAwait(false));
+                throw new ArgumentNullException(nameof(text));
             }
 
-            ValidateRequest(httpRequest, body, validationUrl, authToken);
+            if (encoding == null)
+            {
+                throw new ArgumentNullException(nameof(encoding));
+            }
+
+            response.ContentType = "text/plain";
+            response.StatusCode = code;
+
+            var data = encoding.GetBytes(text);
 
-            var twilioMessage = JsonConvert.DeserializeObject<TwilioMessage>(JsonConvert.SerializeObject(body));
+            await response.Body.WriteAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Creates a Bot Framework <see cref="Activity"/> from an HTTP request that contains a Twilio message.
+        /// </summary>
+        /// <param name="payload">The HTTP request.</param>
+        /// <returns>The activity object.</returns>
+        public static Activity PayloadToActivity(Dictionary<string, string> payload)
+        {
+            if (payload == null)
+            {
+                return null;
```

As per the advice on the other adapters (confirmed with Chris / Gabo) please throw an exception here instead of returning null.